### DISCUSSION TITLE
Can you configure distributed cache as only cache store?

### DISCUSSION THREAD

**Discussion by vsudhini** [ZiggyCreatures/FusionCache#165] [Q&A] [answered]
For certain cache items that are large in size, want to persist them in distributed cache store only. 
Do not want the large cache objects in memory because of the limited amount of resources available in share cloud infrastructures.
Is there a way to configure fusioncache achieve this maybe using named caching?

**Comment by jodydonetti** [comment]
Hi @vsudhini and thanks for using FusionCache!

To skip the memory cache you can use the `SkipMemoryCache` option, see [here](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/Options.md) for more info on all the available options.

As with all other entry options in FusionCache, you can set it once for the entire cache in the `DefaultEntryOptions` or you can set it for each call, for a more granular control.

⚠️ Keep in mind though that without a memory cache you'll basically loose the [Cache Stampede](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/CacheStampede.md) prevention, because without a memory cache that is impossible to achieve.

Let me know if this works for you.

Hope this helps.

**Reply by vsudhini** [reply]
@jodydonetti thanks for responding so promptly.
I have seen cache tower lib dealing with cache stampedes when using distributed store (Redis) as mentioned here https://github.com/TurnerSoftware/CacheTower#distributed-locking-via-redis
Not sure if this will apply to fusioncache.

**Reply by vsudhini** [reply]
Will try the skip memory cache option 🙏🏼

**Reply by jodydonetti** [reply, marked as answer]
> I have seen cache tower lib dealing with cache stampedes when using distributed store (Redis) as mentioned here
> https://github.com/TurnerSoftware/CacheTower#distributed-locking-via-redis
> Not sure if this will apply to fusioncache.

Hi, I should've explained myself more clearly: the Cache Stampede prevention against the database is still there (so 1.000 requests coming in for the same piece of data would go to the database at most once), but not for the distributed cache itself.

This means that all the requests would go to the distirbuted cache (but, again, not to the DB) since without a local memory cache there would be no local state, so every request will need to read from the distributed cache itself, everytime.

If you use CacheTower with only the distributed cache (no memory cache) + the distributed locking extension, you'll still go to the distributed cache every single time.

If this is what you want, you can get the same result with FusionCache via the `SkipMemoryCache` option.

The only remaining difference would be that with CacheTower, in a multi-node environment, only 1 node would go to the database, whereas with FusionCache more than 1 node may do that.

Hope this helps.

**Reply by vsudhini** [reply]
That helps a lot, it’s a trade off that we can live with.
Thank you again.
